test/e2e/util.go

```diff
@@ -50,29 +56,25 @@ func Failf(format string, a ...interface{}) {
 	Fail(fmt.Sprintf(format, a...), 1)
 }
 
-func waitForPodRunning(c *client.Client, id string, tryFor time.Duration) error {
-	trySecs := int(tryFor.Seconds())
-	for i := 0; i <= trySecs; i += 5 {
-		time.Sleep(5 * time.Second)
-		pod, err := c.Pods(api.NamespaceDefault).Get(id)
+func waitForPodRunning(c *client.Client, podName string) error {
+	By(fmt.Sprintf("waiting up to %v for pod %s status to be running", podStartTimeout, podName))
+	for start := time.Now(); time.Since(start) < podStartTimeout; time.Sleep(5 * time.Second) {
+		pod, err := c.Pods(api.NamespaceDefault).Get(podName)
 		if err != nil {
-			return fmt.Errorf("Get pod %s failed: %v", id, err.Error())
+			return fmt.Errorf("Get pod %s failed: %v", podName, err.Error())
```

Comment:
If there is an error here should we log it and retry rather than aborting before the timeout has been reached?